src/models/event.js

```diff
@@ -299,6 +310,100 @@ utils.extend(module.exports.MatrixEvent.prototype, {
     },
 
     /**
+     * Check if this event is currently being decrypted.
+     *
+     * @return {boolean} True if this event is currently being decrypted, else false.
+     */
+    isBeingDecrypted: function() {
+        return this._decrypting;
+    },
+
+    /**
+     * Start the process of trying to decrypt this event.
+     *
+     * (This is used within the SDK: it isn't intended for use by applications)
+     *
+     * @internal
+     *
+     * @param {module:crypto} crypto crypto module
+     */
+    attemptDecryption: function(crypto) {
+        if (!crypto) {
+            this._badEncryptedMessage("Encryption not enabled");
+            return;
+        }
+
+        if (!this.isEncrypted()) {
+            throw new Error("Attempt to decrypt event which isn't encrypted");
```

Comment:
There being no crypto module is an expected condition. The app is not usable; the user can use it as normal in unencrypted rooms. Encrypted messages are shown as "Cannot decrypt: Encryption not enabled".

`attemptDecryption` being called on an event which is not encrypted is not an expected condition; it means something has gone wrong somewhere; the test here is just for robustness.

What would you like me to do?